biggus/__init__.py

```diff
@@ -561,6 +563,175 @@ def _normalise_keys(self, keys):
         return keys
 
 
+class ArrayContainer(Array):
```

Comment:
I would have gone for either not introducing `ArrayContainer` in this PR at all, or introducing it and refactoring the other classes at the same time.